File: content/courses/parallel-computing-introduction/codes/bcast.cxx
#include <iostream>
#include <iomanip>
#include "mpi.h"

using namespace std;

int main(int argc, char *argv[]) {

    int nprocs,rank;

    MPI_Init(&argc,&argv);
    MPI_Comm_size(MPI_COMM_WORLD,&nprocs);
    MPI_Comm_rank(MPI_COMM_WORLD,&rank);

    //extra spece for null terminator
    double message;
    if (rank==0) {
        message=42.;
    }

    MPI_Bcast(&message,1,MPI_DOUBLE,0,MPI_COMM_WORLD);

    cout<<rank<<" "<<message<<endl;

    MPI_Finalize();

}
        

File: content/courses/parallel-computing-introduction/codes/bcast.f90
program broadcaster
    use mpi
    implicit none

    integer           :: nprocs, rank, ierr
    double precision  :: message

    call MPI_Init(ierr)
    call MPI_Comm_size(MPI_COMM_WORLD,nprocs,ierr)
    call MPI_Comm_rank(MPI_COMM_WORLD,rank,ierr)

    if (rank==0) then
        message=42.
    endif

    call MPI_Bcast(message,1,MPI_DOUBLE_PRECISION,0,MPI_COMM_WORLD,ierr)

    print*, rank, message

    call MPI_Finalize(ierr)

end program 
